ydb/core/persqueue/events/internal.h

```diff
@@ -1076,6 +1077,15 @@ struct TEvPQ {
         {
         }
     };
+
+    struct TEvReadingPartitionFinishedRequest : public TEventPB<TEvReadingPartitionFinishedRequest, NKikimrPQ::TEvReadingPartitionFinishedRequest, EvReadingPartitionFinishedRequest> {
```

Comment:
Make ReadPartitionStatusRequest with Status=Finished|NotFinished. You will use the same event when client will recreate read-rule or commit to the past.